### PR TITLE
add client and tcp_client_transport

### PR DESCRIPTION
**Pull request by oleg1**

### CODE REVIEW

File: lib/transports/TCP.js [hunk 1]
```diff
@@ -247,4 +247,102 @@ Transport.prototype.close = function () {
...
+
+    self.connection.on('close', function () {
+        console.log('c -> rcvd close');
+        if (!self.closed) {
+            self.closed = true;
+            self.emit('close');
+        }
+    });
+}
+util.inherits(ClientTransport,ClientTransportPrototype);
```
Comment (mcheshkov) [inline lib/transports/TCP.js:280]: You sure this works, and not `ReferenceError: ClientTransportPrototype is not defined` ?

Comment (oleg1) [inline lib/transports/TCP.js:280]: fixed

File: lib/transports/TCP.js [hunk 2]
```diff
@@ -117,35 +119,33 @@ Server.prototype._onConnection = function(conn){
...
+	var CIDMarker = null;
+    var gotCID = false;
+    var cid;
+    var oc = new Connection();
 
     function onData(dataReceived){
         data = Buffer.concat([data,dataReceived],data.length + dataReceived.length);
-		if (!gotHelo && policyRegExp.test(data.toString())){
-			// Посылаем XML и закрываем сокет:
-			conn.end(self._crossdomainXML);
```
Comment (mcheshkov) [inline lib/transports/TCP.js:129]: Now nobody will ever send crossdomain policy, and flash won't connect

File: lib/transports/TCP.js [hunk 3]
```diff
@@ -247,4 +247,102 @@ Transport.prototype.close = function () {
...
+        } 
+          
+        if(!gotHelo) {
+            // Если HELO подошел мы должны обрезать данные, чтобы передать их дальше.
+            data = data.slice(helo_bl);
+            gotHelo = true;
+        }     
+
+        if (!gotCIDMarker && data.length === 0) return;        
+        if(CIDMarker === null) {
```
Comment (mcheshkov) [inline lib/transports/TCP.js:328]: Client transport should write CID marker, so server transport could read it. Now they both read it.

File: lib/transports/TCP.js (hunk 2 above)
Comment (oleg1) [inline lib/transports/TCP.js:129]: fixed

File: lib/transports/TCP.js (hunk 3 above)
Comment (oleg1) [inline lib/transports/TCP.js:328]: fixed

**Pull request merged**